src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -176,18 +185,47 @@ private static Dataset makeDataset(CloudPlatform cloudPlatform) {
         .id(UUID.randomUUID());
   }
 
-  private void mockRunQueryForConcepts(
+  private void mockRunQueryForSearchConcepts(
       CloudPlatform cloudPlatform, List<SnapshotBuilderConcept> concepts, Dataset dataset) {
     CloudPlatformWrapper cloudPlatformWrapper = CloudPlatformWrapper.of(cloudPlatform);
-    if (cloudPlatformWrapper.isGcp()) {
-      when(datasetService.retrieveModel(dataset, TEST_USER)).thenReturn(makeDatasetModel());
-      when(bigQueryDatasetPdao.<SnapshotBuilderConcept>runQuery(any(), any(), any()))
-          .thenReturn(concepts);
-    } else {
-      when(datasetService.getOrCreateExternalAzureDataSource(dataset, TEST_USER))
-          .thenReturn("dataSource");
-      when(azureSynapsePdao.<SnapshotBuilderConcept>runQuery(any(), any())).thenReturn(concepts);
-    }
+    cloudPlatformWrapper.choose(
+        () -> {
+          when(datasetService.retrieveModel(dataset, TEST_USER)).thenReturn(makeDatasetModel());
+          when(bigQueryDatasetPdao.<SnapshotBuilderConcept>runQuery(any(), any(), any()))
+              .thenReturn(concepts);
+          return null;
```

Comment:
Instead of `return null` you could make these `choose()` functions return the stub. Something like
```
var stub = choose(() -> { ... return when(bigQueryDatasetPdao.<SnapshotBuilderConcept>runQuery(any(), any(), any())); } ... );
stub.thenReturn(concepts);
```

If you refactor this way, then I think you can combine most of `mockRunQueryForSearchConcepts` and `mockRunQueryForGetConcepts` as the only difference would be the returned value passed to `thenReturn`.